development/image-syncer/util.go

```diff
@@ -17,6 +17,9 @@ func getTarget(source, targetRepo, targetTag string) (string, error) {
 		}
 		imageName := strings.Split(source, "@sha256:")[0]
 		target = targetRepo + imageName + ":" + targetTag
+	} else if targetTag != "" {
```

Comment:
Please add some brief comment why we have this condition check here. Something like "Allow retagging when the source image is not using SHA256".